### PR TITLE
logic for firefox detection moved to ember-metal/environment

### PR DESCRIPTION
**Pull request by mateusvahl**

### CODE REVIEW

File: packages/ember-metal/lib/environment.js [hunk 1]
```diff
@@ -34,6 +35,7 @@ if (hasDOM) {
   environment = {
     hasDOM: false,
     isChrome: false,
+    isFirefox: false
```
Comment (rwjblue) [inline packages/ember-metal/lib/environment.js]: This is missing a trailing comma.

**Comment by rwjblue** [comment]
Can you rebase and squash the commits?

**Comment by mateuspv** [comment]
Mess, apparently I can you rebase and squash the commits...

**Comment by mateuspv** [comment]
@rwjblue Should I rebase and squash one more time(correctly now)?

**Comment by rwjblue** [comment]
Yes, please

**Comment by rwjblue** [comment]
rebase & squash please?

**Comment by mateuspv** [comment]
@rwjblue could you close? I can't figure out ^^

**Comment by stefanpenner** [comment]
i got it

**Comment by stefanpenner** [comment]
https://github.com/emberjs/ember.js/pull/10820 i believe should doit, @mateuspv mind confirming i didn't screw anything up?

**Pull request closed without merging**